### ISSUE TITLE
Freezing during account setup

### ISSUE THREAD

**Issue by dilyanpalauzov** [jobisoft/DAV-4-TbSync#178]
When I setup an anonymous account (only host name without username) for host milter.aegee.org I see an event in the calendar.  When I setup another hostname-only account for k.bapha.be . TbSync shows the calendars, obtains the different report types, but does not show the event and does not perform REPORT to get the events.  Why, what is the difference?

Moreover when I setup account for anonymous@bapha.be with any password, TbSync calls PROPFIND for the calendars, but then freezes and makes no calls for the addressbooks.  It is not possible to unfreeze TB.

If I setup account for user anonymous@milter.aegee.org without explicit host and any password  it just works. So what is the difference?

For reference DAVx5 and Gnome Evolution complete without problems and show the event behind  bapha.be.

**Comment by jobisoft** [comment]
You are running into bug:
https://bugzilla.mozilla.org/show_bug.cgi?id=1544596

Without username, Lightning cannot put both in different containers. Nothing I can do.

For the freeze it looks like TbSync died on a javascript error, which should be in the log. Are you tzalking about the setup wizard?

**Comment by dilyanpalauzov** [comment]
I am talking about the setup wizzard.

I still do not understand why milter.aegee.org and k.bapha.be behave differently.

**Comment by dilyanpalauzov** [comment]
I do not think that  bug 1544596 is related. In particular, if I enter as user anonymous@bapha.be with any password there is username from TBs perspective.  So there is no justification for freezing.

**Comment by jobisoft** [comment]
You are right, I mixed something up. I thought anonymous means no username given. I am looking into it.

**Comment by jobisoft** [comment]
For k.bapha.be you are running into a lighning bug. The calendar discovered is:
https://k.bapha.be/dav/calendars/user/tls@bapha.be/TLS/

And if you set that up manually it will crash as well. I hav no resource to investigate that.
![grafik](https://user-images.githubusercontent.com/5830621/69655776-97135280-1077-11ea-8d66-5a41c160a0b1.png)

**Comment by jobisoft** [comment]
More details: If you add a username for the above calendar, it works. Only if you run fully anonymous it dies.

**Comment by jobisoft** [comment]
>Moreover when I setup account for anonymous@bapha.be with any password, TbSync calls PROPFIND for the calendars, but then freezes and makes no calls for the addressbooks. It is not possible to unfreeze TB.

I can not confirm this with the latest beta.
![grafik](https://user-images.githubusercontent.com/5830621/69656434-d68e6e80-1078-11ea-911c-05b07a9381ca.png)

I can finish this setup and can load the calendar. The calendar dies with the aString error.

**Comment by dilyanpalauzov** [comment]
On my system with the beta addons, the setup wizzard freezes. It calls the calenders, but not the addressbooks.

The point is, that if the setup does complete, no REPORT is issued to fetch the events, but the contacts are fetched with REPORT.

Why for anonymous@bapha.be the developer console (Network) prints, that it tried first insecure connection?

**Comment by dilyanpalauzov** [comment]
I filled https://bugzilla.mozilla.org/show_bug.cgi?id=1599541 .

**Comment by jobisoft** [comment]
> Why for anonymous@bapha.be the developer console (Network) prints, that it tried first insecure connection?

Cannot confirm that.

> I filled https://bugzilla.mozilla.org/show_bug.cgi?id=1599541 .

The way you phrased that, it will probably be rejected. You need to reproduce the issue without using TbSync as I did here:
https://github.com/jobisoft/DAV-4-TbSync/issues/178#issuecomment-558727062

**Comment by dilyanpalauzov** [comment]
How can I debug the cause of thr freezing?

**Comment by dilyanpalauzov** [comment]
I have to say that the setup sometimes completes and sometimes it does not complete.  When I start account setup, TbSync discovers the collections, then I delete the account and start a new setup, the wizzard freezes and does not make any WebDAV requests

**Comment by jobisoft** [comment]
You are still talking about that special account which does not have an Scheduling-INBOX? Or is it a general issue? I have not observed that.

**Comment by dilyanpalauzov** [comment]
It is a geneal issue.  I setup account for the user aaa@aegee.org with passoword  S, which is wrong.  The developer console shows to PROPFINDs on https://mail.aegee.org/dav/calendars, on which the server returns 401 (as the password is wrong).  The setup wizzard freezes, the process cannot be Cancelled. The *Back* button is greyed out.  Only *killall* helps.

However, sometimes it completes.  I really do not understand why the behaviour is not consistent.  I use TB 68.2.2 on Linux/Debian.

**Comment by dilyanpalauzov** [comment]
In this setup I do not enter server name.

**Comment by jobisoft** [comment]
It looks like your server is looping redirects. A request for `https://www.aegee.org/.well-known/carddav` gets a 301 to the same location.

![grafik](https://user-images.githubusercontent.com/5830621/70282918-b02aaa80-17bf-11ea-9837-2c4c2024b8f2.png)

Same with caldav.

**Comment by dilyanpalauzov** [comment]
As a matter of fact for AEGEE.ORG the server shall be obtained from DNS SRV/TXT _caldavs._tcp.aegee.org and _carddavs._tcp.aegee.org .  The server behing (www.)aegee.org redirects from an address carddav/ to an address terminating with slash (carddav/) and then unfortunately is says PROPFIND not implemented.  But this address shall be used only after the DNS lookups have failed.

**Comment by dilyanpalauzov** [comment]
What ever the WebDAV server does, the setup wizzard shall not freeze and the Cancel and Back buttons must work.

**Comment by jobisoft** [comment]
So you send wrong details when a client is not doing SRV lookup? Why?

Is there a defined order of SRV vs well-known?

**Comment by jobisoft** [comment]
It looks like you changed something on your server again. I checked my code and I do the DNS lookup first and for aaa@aegee.org I get 

```
SRV record @ _carddavs._tcp.aegee.org
{"prio":0,"weight":1,"host":"mail.aegee.org","port":443,"type":33,"nsclass":0,"ttl":49513}
```
and from `https://mail.aegee.org/.well-known/carddav` I get redirected to `https://mail.aegee.org/dav/addressbooks` which fails with 401 (as S is the wrong password). No freezing.

I may add redirect loop protection but your server is not doing that anymore, so nothing to test against.

**Comment by dilyanpalauzov** [comment]
The webserver behind www.aegee.org is running Apache HTTPD and I was told that this server cannot be configured to answer on PROPFIND requests with redirect (GET would redirect).  I do not run that server.

I do not see a loop.  As I said, sometimes the procedure completes and sometimes the window freezes, with non-reacting Cancel button.  It is a matter of repeating the procedure to reach frozen state.

After fetching the SRV record from DNS, the software shall fetch the TXT record at the same address and construct from it the stub URL. The described workflow above mentions that the server is taken from DNS SRV, but the then /.well-known/carddav is tried.  This means, that TXT is not considered.

The bootsrtaping procedure by https://tools.ietf.org/html/rfc6764#section-6 is first SRVand if it fails, well-known.  A good implementation of the standards suggests reading and understanding the standards.

**Comment by jobisoft** [comment]
>A good implementation of the standards suggests reading and understanding the standards.

I really do not like your attitude and it is pulling me down. I will not respond to your requests anymore.

I implemented a lot of your feature requests, but there has never been a word of appreciation. All you do is complain. I am no expert in none of this and I do that in my free time and I did the best I can.

Analysing what your server does takes a lot of time (and I spend a bunch of hours working with your servers) and sometimes I forget things. I DO querry the TXT record but I simply did not log it to the event recorder and thus assumed it was a standard redirect (which i veriefied with an online redirect tool, for GET of course).

The redirects (and the loop) does not happen anymore since you changed your server config yesterday.

Good bye.
